pkg/spanconfig/testsqlwatcher/test_reconciler.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package testsqlwatcher
```

Comment:
Get rid of this package and implement a mock sqlwatcher in the test that wants to pass in a fake one. It will only happen in the spanconfigmanager package, so we don't need a full package for this.